certbot-apache/certbot_apache/configurator.py

```diff
@@ -202,6 +202,11 @@ def __init__(self, *args, **kwargs):
         self._autohsts = {}  # type: Dict[str, Dict[str, Union[int, float]]]
         # Reverter save notes
         self.save_notes = ""
+        # Should we use ParserNode implementation instead of the old behavior
+        self.USE_PARSERNODE = False
+        # Saves the list of file paths that were parsed initially, and
```

Comment:
We're using "parsed initially" here. I assume that means "on augeas' initial passthrough". I assume the intent is to figure out which ones are enabled in the config that will actually be loaded by apache. In that case, we should check that we're setting things appropriately anytime we do some change that might modify this -- essentially, making sure it's set any time we do a full passthrough, whether by augeas or apacheparser.

Reply:
You are right. At the current state with Augeas parser maintaining a meaningful value here gets a bit cumbersome, and hence only the very first initial parse state is maintained for this purpose. Some of the reasons it hasn't historically been feasible to maintain the state for this variable are:
 - Autosaving anything not-yet-written-to-disk on `find_dir` as it also looks for new `Include` and `IncludeOptional` directives - causing a reload of Augeas parsing tree, and erasing all unsaved changes.
 - "Manually" added parse paths to include `VirtualHost` files from arbitrary location using `--apache-vhost-path`

This  should be changed in the future, but for the sake of sanity during the migration process, should probably be implemented in the current, naive way.

Reply:
As we discussed offline, agreed to leave this for now and possibly fix once we've moved off of augeas.

Reply:
As we discussed offline, agreed to leave this for now and possibly fix once we've moved off of augeas.